From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Wed, 13 Apr 2022 16:17:23 +0800
Subject: kms/crtc: Add debug logging of supported vs unsupported properties

Unsupported properties are particularly common in an Nvidia/hybrid
setup so we don't want to make it a warning that's always visible.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2360>
Origin: upstream, 42.1, commit:cc20584032faff174f2d924464617a68661516e4
---
 src/backends/native/meta-kms-crtc.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/backends/native/meta-kms-crtc.c b/src/backends/native/meta-kms-crtc.c
index 6706e16..1ae3d84 100644
--- a/src/backends/native/meta-kms-crtc.c
+++ b/src/backends/native/meta-kms-crtc.c
@@ -369,6 +369,7 @@ init_properties (MetaKmsCrtc       *crtc,
   MetaKmsCrtcPropTable *prop_table = &crtc->prop_table;
   int fd;
   drmModeObjectProperties *drm_props;
+  int i;
 
   *prop_table = (MetaKmsCrtcPropTable) {
     .props = {
@@ -405,6 +406,17 @@ init_properties (MetaKmsCrtc       *crtc,
                                         crtc);
 
   drmModeFreeObjectProperties (drm_props);
+
+  for (i = 0; i < META_KMS_CRTC_N_PROPS; i++)
+    {
+      meta_topic (META_DEBUG_KMS,
+                  "%s (%s) CRTC %u property '%s' is %s",
+                  meta_kms_impl_device_get_path (impl_device),
+                  meta_kms_impl_device_get_driver_name (impl_device),
+                  drm_crtc->crtc_id,
+                  prop_table->props[i].name,
+                  prop_table->props[i].prop_id ? "supported" : "unsupported");
+    }
 }
 
 MetaKmsCrtc *
